evap/staff/importers/json.py

```diff
@@ -0,0 +1,417 @@
+            # Update previously created main evaluation
+            course.evaluations.all().update(
+                wait_for_grade_upload_before_publishing=wait_for_grade_upload_before_publishing
+            )
+        else:
+            # Set evaluation time frame of two weeks for normal evaluations:
+            # Start datetime is at 8:00 am on the monday in the week before the event ends
+            evaluation_start_datetime = course_end.replace(hour=8, minute=0, second=0, microsecond=0) - timedelta(
+                weeks=1, days=course_end.weekday()
+            )
+            # End date is on the sunday in the week the event ends
+            evaluation_end_date = (course_end + timedelta(days=6 - course_end.weekday())).date()
+
+            name_de, name_en = "", ""
+
+            weight = 9
+
+            # Might be overwritten when importing related exam evaluation
+            wait_for_grade_upload_before_publishing = True
+
+        participants = self._get_user_profiles(data["students"]) if "students" in data else []
+
+        defaults = {
+            "name_de": name_de,
+            "name_en": name_en,
+            "vote_start_datetime": evaluation_start_datetime,
+            "vote_end_date": evaluation_end_date,
+            "wait_for_grade_upload_before_publishing": wait_for_grade_upload_before_publishing,
+            "weight": weight,
+        }
+        evaluation, created = Evaluation.objects.get_or_create(
+            course=course,
+            cms_id=data["gguid"],
+            defaults=defaults,
+        )
+        if evaluation.state < Evaluation.State.APPROVED:
+            direct_changes = update_with_changes(evaluation, defaults)
+
+            participant_changes = set(evaluation.participants.all()) != set(participants)
+            evaluation.participants.set(participants)
+
+            any_lecturers_changed = False
+            if "lecturers" not in data:
+                self.statistics.warnings.append(
+                    WarningMessage(obj=evaluation.full_name, message="No contributors defined")
+                )
+            else:
+                for lecturer in data["lecturers"]:
+                    __, lecturer_created, lecturer_changes = self._import_contribution(evaluation, lecturer)
+                    if lecturer_changes or lecturer_created:
+                        any_lecturers_changed = True
+
+            if direct_changes or participant_changes or any_lecturers_changed:
+                self.statistics.updated_evaluations.append(evaluation)
+        else:
+            self.statistics.attempted_changes.append(evaluation)
+
+        if created:
+            self.statistics.new_evaluations.append(evaluation)
+
+        return evaluation
+
+    def _import_contribution(
+        self, evaluation: Evaluation, data: ImportRelated
+    ) -> tuple[Contribution | None, bool, dict[str, tuple[any, any]]]:
+        if data["gguid"] not in self.user_profile_map:
+            return None, False, {}
```

Comment:
Do we expect this to happen? If yes, in what case? Can we add a comment here and a test?

Reply:
@janno42 Do we expect this to happen?

Reply:
students and lecturers without email are skipped (see `_import_students` and `_import_lecturers`), so they are not in the list.